compiler/rustc_const_eval/src/transform/check_consts/check.rs

```diff
@@ -129,6 +129,8 @@ impl<'mir, 'tcx> Qualifs<'mir, 'tcx> {
         ccx: &'mir ConstCx<'mir, 'tcx>,
         tainted_by_errors: Option<ErrorGuaranteed>,
     ) -> ConstQualifs {
+        // FIXME(explicit_tail_calls): uhhhh I think we can return without return now, does it change anything
```

Comment:
Afaict we don't run `mir_const_qualif` on `const fn` anymore (we should change it to assert that instead of silently returning the `Default` for functions and actually running on `const fn` were it called for them), so you'd only encounter this situation in constants. 😆 do you have tests for `const FOO: u32 = 42; const BAR: u32 = become FOO;`?

Reply:
I don't have a test for it, yet, will add.


To be clear, the following makes an error like "become outside of function body" (similarly to how `return`) works:
```rust
const FOO: u32 = 42;
const BAR: u32 = become FOO;
```

And this ICEs in this PR, but will produce "`become` requires a function call" once we have the checks (I'll make a PR with them soon in parallel with this one):
```rust
const FOO: u32 = 42;

fn bar() -> u32 {
    become FOO;
}
```